examples/simple/BUILD

```diff
@@ -13,9 +13,15 @@ swift_binary(
     visibility = ["//visibility:public"],
 )
 
+filegroup(
+    name = "pre_build",
+    srcs = ["pre build.sh"],
+)
+
 xcodeproj(
     name = "xcodeproj",
     build_mode = "bazel",
+    pre_build = ":pre_build",
```

Comment:
This wouldn't make it possible to have different pre-build scripts for different targets, right? For example, we have one test target where we'd like to fetch some big outputs required for the tests to run, but for other top-level targets we'd like to run SwiftLint for example. My initial idea for this implementation was to have a dictionary `target -> script` like I recently added for `extra_files`, but maybe that's not needed in this case.

Reply:
While this approach is similar to what we used in our last project generator and solves for some basic things like swiftlint integration, I agree that being able to apply this at the target level is important long term.

How about:

- Making this a label_list instead of a label, for multiple pre/post build scripts
- Each item is a, let's call it `xcodeproj.buildphasescript` struct with a label (for the script) and an optional label_list that calls out the targets that script applies to (similar utility to what @BalestraPatrick mentioned)

That way the easy/default way is to apply any script to all targets, but each script can target specific targets if needed.

Reply:
Thanks @BalestraPatrick and @erikkerber for the comments!

The suggestions all make sense. I have a question: where (which targets) should we insert the pre-build phases? In BwB, the actual build happens in the `BazelDependencies` target, the build in each library target is just a fake build. If we insert the pre-build phase in the library target, it would only run when the actual build already finishes. But if we insert it in the `BazelDependencies` target, we'd have to always generate a `BazelDependencies` target even in BwX (AFAIK `BazelDependencies` is not generated if everything can be built with Xcode).

> we have one test target where we'd like to fetch some big outputs

Reply:
That's a good point @thii, and in the Swiftlint specific case this could just as easily live in a build phase for `BazelDependencies` itself (and not on every target, or any combination of other targets).

To that point it's worth flushing the use cases out. I didn't realize `BazelDependencies` would be omitted if not needed, which makes sense. 

Depending on hearing more about @BalestraPatrick's use case, it could be worth always generating `BazelDependencies` for BwX while not always including the Bazel script itself if it's not needed while including any `pre_build` scripts. This would suite the "insert tools that generate warnings and errors into Xcode" use case, but would remove any real `post_build` scripts since things would happen pre-build for most of BwX.

Reply:
>I didn't realize BazelDependencies would be omitted if not needed, which makes sense.

But "if needed" depends on various things, and we can extend it to include pre-build scripts in BwX, without forcing it to also include a `bazel_build.sh` call.